instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/rules/SessionReplayTestRule.kt

```diff
@@ -21,6 +21,12 @@ internal open class SessionReplayTestRule<T : Activity>(
     // region ActivityTestRule
 
     override fun beforeActivityLaunched() {
+        // give time to remove the previous activity. Espresso seems to have moments
+        // when it tries to launch the new activity while the previous one is still somehow
+        // in the process of being removed. This creates an issue with our SR recorder which
+        // calls the WindowInspector.getGlobalWindowViews() which can return the previous window +
+        // the current window and alters the tests.
+        Thread.sleep(2000)
```

Comment:
Interesting thing is that `ActivityTestRule` we are using is [deprecated](https://developer.android.com/reference/androidx/test/rule/ActivityTestRule) and the proposed replacement is `ActivityScenarioRule` which have a better control over the lifecycle by providing `moveToState` blocking methods. Maybe we should consider the possibility of migrating to `ActivityScenarioRule` later.